packages/suite/src/components/suite/modals/SignedMessage/index.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { createRef, useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Switch, Button, Modal, colors, variables } from '@trezor/components';
+import { Translation } from '@suite-components';
+import * as notificationActions from '@suite-actions/notificationActions';
+import { useActions } from '@suite-hooks';
+import { SectionItem, ActionColumn, TextColumn } from '@suite-components/Settings';
+import { copyToClipboard } from '@suite-utils/dom';
+
+const LogWrapper = styled.pre`
+    padding: 20px;
+    height: 400px;
+    width: 100%;
+    overflow: auto;
+    background-color: ${colors.BLACK96};
+    color: ${colors.BLACK0};
+    font-size: ${variables.FONT_SIZE.TINY};
+    text-align: left;
+    word-break: break-all;
+`;
+
+const ButtonWrapper = styled.div`
+    display: flex;
+    width: 100%;
+    justify-content: space-between;
+`;
+
+type Props = {
+    onCancel: () => void;
+    signedObj: {
+        address: string;
+        message: string;
+        signature: string;
+    };
+};
+
+const SignedMessage = ({ signedObj, onCancel }: Props) => {
+    const { addNotification } = useActions({
+        addNotification: notificationActions.addToast,
+    });
+
+    const htmlElement = createRef<HTMLPreElement>();
+    const prettifyLog = (json: Record<any, any>) => {
+        return { text: JSON.stringify(json, null, 2), isStringifiedJSON: true };
+    };
+    const [signedTextObj, setSignedTextObj] = useState(prettifyLog(signedObj));
+
+    const toggleToPlainText = () => {
+        if (signedTextObj.isStringifiedJSON) {
+            setSignedTextObj({
+                text: `Address: ${signedObj.address}\nMessage: ${signedObj.message}\nSignature: ${signedObj.signature}`,
+                isStringifiedJSON: false,
+            });
+        } else {
+            setSignedTextObj(prettifyLog(signedObj));
+        }
+    };
+
+    const copy = () => {
+        const result = copyToClipboard(signedTextObj.text, htmlElement.current);
+        if (typeof result !== 'string') {
+            addNotification({ type: 'copy-to-clipboard' });
+        }
+    };
+
+    useEffect(copy, []); // componentDidMount, copy at first appear
+
+    const download = () => {
```

Comment:
Instead of copy/pasting this function from `Log` modal move this to `@suite-utils/dom`, right next to `copyToClipboard`
and then use it in both places, here and `Log` modal

Reply:
I did not consider it necessary to rewrite someone else's component, which I do not know exactly where it is used and when, I have been on this project for less than a month, you now wrote - OK, I can do refactoring